store/storethehash/storethehash.go

```diff
@@ -66,14 +70,66 @@ func (s *sthStorage) get(k []byte) ([]indexer.Value, bool, error) {
 		return nil, false, nil
 	}
 
-	out, err := indexer.Unmarshal(value)
+	out, err := indexer.UnmarshalValues(value)
 	if err != nil {
 		return nil, false, err
 	}
 	return out, true, nil
 
 }
 
+func (s *sthStorage) ForEach(iterFunc indexer.IterFunc) error {
+	// It is necessary to do both flush and sync before creating an iterator to
+	// read values that have not been written to file yet
+	s.primary.Flush()
```

Comment:
I think this line is redundant. `Sync()` already flushes the primary storage: https://github.com/ipld/go-storethehash/blob/d72ca9b0968c52b249dc13e901757f1aafcbd5eb/store/store.go#L331

Reply:
I thought that also, but apparently not.  Flush is needed to do commit, and sync is needed to write to file.

Reply:
Oh weird! From the code I would have expected that `Sync` does this for you. Then don't mind me.